model/src/main/java/org/jboss/pnc/model/TargetRepository.java

```diff
@@ -0,0 +1,160 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.Set;
+
+/**
+ * Contains information related to a repository of build artifacts (i.e. Maven, NPM, etc)
+ */
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(uniqueConstraints = @UniqueConstraint(columnNames = { "identifier", "repositoryPath" }) )
+public class TargetRepository implements GenericEntity<Integer> {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final String SEQUENCE_NAME = "artifact_repo_id_seq";
+
+    @Id
+    @Getter
+    @Setter
+    @SequenceGenerator(name = SEQUENCE_NAME, sequenceName = SEQUENCE_NAME, allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = SEQUENCE_NAME)
+    private Integer id;
+
+    /**
+     * Identifier to link repository configurations (eg. hostname)
+     */
+    @Getter
+    @Setter
+    @NotNull
+    @Size(max=255)
+    @Column(updatable=false)
+    private String identifier;
+
+    /**
+     * Path that need to be appended to the hostname
+     * eg. "ga" for https://maven.repository.redhat.com/ga/
+     * or "maven2" for https://repo1.maven.org/maven2/
+     * or "" (empty string) when the repository content starts at root
+     */
+    @Getter
+    @Setter
+    @NotNull
+    @Size(max=255)
+    @Column(updatable=false)
+    private String repositoryPath;
+
+
+    /**
+     * The type of repository which hosts this artifact (Maven, NPM, etc).  This field determines
+     * the format of the identifier string.
+     */
+    @Getter
+    @Setter
+    @NotNull
+    @Column(updatable=false)
+    @Enumerated(EnumType.STRING)
+    private TargetRepository.Type repositoryType;
+
+    @OneToMany(mappedBy = "targetRepository")
+    private Set<Artifact> artifacts;
+
+    /**
+     * Types of artifact repositories
+     *
+     * Types *_TEMPORAL are repository groups for temporal build (Snapshots / Pull Request)
+     */
+    public enum Type {
+        /**
+         * Maven artifact repository such as Maven central (http://central.maven.org/maven2/)
+         */
+        MAVEN,
+        MAVEN_TEMPORAL,
```

Comment:
I think that this name is not ideal as it means "related to time" :-). Temporal databases are used to store time related data. The work temporary is not ideal, but I'd say better.